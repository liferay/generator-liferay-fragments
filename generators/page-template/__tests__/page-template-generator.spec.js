const fs = require('fs');
const path = require('path');
const YeomanTest = require('yeoman-test');

/**
 * @param  {string[]} paths
 */
function expectFile(...paths) {
  return expect(
    fs.readFileSync(path.join(...paths), 'utf-8')
  ).toMatchSnapshot();
}

/**
 * @param  {string} base
 * @param  {string[]} paths
 */
function expectFiles(base, paths) {
  return paths.map(_path => expectFile(path.join(base, _path)));
}

describe('page-template-generator', () => {
  it('generates a new display page template', () =>
    YeomanTest.run(path.join(__dirname, '..'))
      .withOptions({ pageTemplateName: 'Sample Display Page Template' })
      .withOptions({ pageTemplateType: 'display-page-template' })
      .then(projectPath => {
        expectFiles(path.join(projectPath, 'src', 'sample-display-page-template'), [
          'display-page-template.json',
          'page-definition.json'
        ]);
      }));

  it('generates a new master page', () =>
    YeomanTest.run(path.join(__dirname, '..'))
      .withOptions({ pageTemplateName: 'Sample Master Page' })
      .withOptions({ pageTemplateType: 'master-page' })
      .then(projectPath => {
        expectFiles(path.join(projectPath, 'src', 'sample-master-page'), [
          'master-page.json',
          'page-definition.json'
        ]);
      }));

  it('Generates a new page template', () =>
    YeomanTest.run(path.join(__dirname, '..'))
      .withOptions({ pageTemplateName: 'Sample Page Template' })
      .withOptions({ pageTemplateType: 'page-template' })
      .then(projectPath => {
        expectFiles(path.join(projectPath, 'src', 'sample-page-template'), [
          'page-template.json',
          'page-definition.json'
        ]);
      }));

  it('needs a name', () =>
    new Promise((resolve, reject) =>
      YeomanTest.run(path.join(__dirname, '..'))
        .withOptions({ pageTemplateType: 'master-page' })
        .then(reject)
        .catch(resolve)
    ));
});
