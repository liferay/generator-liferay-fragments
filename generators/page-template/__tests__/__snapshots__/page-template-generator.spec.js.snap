// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page-template-generator Generates a new page template 1`] = `
"{
  \\"name\\" : \\"Sample Page Template\\"
}
"
`;

exports[`page-template-generator Generates a new page template 2`] = `
"{
  \\"pageElement\\" : {
    \\"pageElements\\" : [ {
      \\"definition\\" : {
        \\"fragment\\" : {
          \\"collectionName\\" : \\"Basic Components\\",
          \\"key\\" : \\"BASIC_COMPONENT-heading\\",
          \\"name\\" : \\"Heading\\"
        },
        \\"fragmentConfig\\" : {
          \\"textAlign\\" : \\"left\\",
          \\"bottomSpacing\\" : \\"3\\",
          \\"headingLevel\\" : \\"h1\\",
          \\"textColor\\" : \\"\\"
        },
        \\"fragmentFields\\" : [ {
          \\"id\\" : \\"element-text\\",
          \\"value\\" : {
            \\"text\\" : {
              \\"value_i18n\\" : {
                \\"en_US\\" : \\"This is a page template\\"
              }
            }
          }
        } ]
      },
      \\"type\\" : \\"Fragment\\"
    } ],
    \\"type\\" : \\"Root\\"
  },
  \\"settings\\" : {
    \\"colorSchemeName\\" : \\"01\\",
    \\"themeName\\" : \\"Classic\\"
  }
}"
`;

exports[`page-template-generator generates a new master page 1`] = `
"{
  \\"name\\" : \\"Sample Master Page\\"
}
"
`;

exports[`page-template-generator generates a new master page 2`] = `
"{
  \\"pageElement\\" : {
    \\"pageElements\\" : [ {
      \\"definition\\" : {
        \\"fragmentSettings\\" : {
          \\"unallowedFragments\\" : [ ]
        }
      },
      \\"type\\" : \\"DropZone\\"
    } ],
    \\"type\\" : \\"Root\\"
  },
  \\"settings\\" : {
    \\"colorSchemeName\\" : \\"01\\",
    \\"themeName\\" : \\"Classic\\",
    \\"themeSettings\\" : {
      \\"lfr-theme:regular:show-header\\" : \\"false\\",
      \\"lfr-theme:regular:wrap-widget-page-content\\" : \\"false\\",
      \\"lfr-theme:regular:show-header-search\\" : \\"false\\",
      \\"lfr-theme:regular:show-footer\\" : \\"false\\"
    }
  }
}
"
`;
